### PR TITLE
Issue 1612

### PR DESCRIPTION
**Pull request by koheiw**
This PR is too improve consistency of handling attributes. Only behavior change is `cbind()` allows duplicated features.

### CODE REVIEW

File: tests/testthat/test-docnames.R [hunk 1]
```diff
@@ -26,14 +26,14 @@ test_that("docnames<- works with corpus, tokens and dfm (#987)", {
...
-    dfmat <- as.dfm(data_dfm_lbgexample)
-    dfmat@Dimnames$docs <- seq_len(ndoc(dfmat))
-    expect_identical(
-        dimnames(as.matrix(textstat_simil(dfmat)))[[1]],
-        as.character(1:6)
-    )
-})
+# test_that("dfm docnames cannot be made integer before textstat_simil", {
+#     dfmat <- as.dfm(data_dfm_lbgexample)
+#     dfmat@Dimnames$docs <- seq_len(ndoc(dfmat))
```
Comment (kbenoit) [inline tests/testthat/test-docnames.R]: No one should be doing this, but the check for dimnames being character only happen when something tries to modify the object, as apparently `textstat_simil()` does. This causes a run-time error even though the assignment of the integer dimnames itself violates the class/slot type declaration. A flaw of S4, true, but is it something we could or should trap?

Comment (kbenoit) [reply tests/testthat/test-docnames.R]: Same error with `docnames(dfmat) <- 1:6`

Comment (koheiw) [reply tests/testthat/test-docnames.R]: We could run `set_dfm_dimnames(x) <- dimnames(x)` in `as.dfm()` to covert number to characters but I am not sure if we need to save people who try to destroy their own objects...

Comment (koheiw) [reply tests/testthat/test-docnames.R]: Good idea to coerce input to character in `docnames()`.

**Comment by koheiw** [comment]
Travis is falling to install libraries.

**Comment by kbenoit** [comment]
Looks good, I will approve if/when Travis finishes the checks. :+1:

**Comment by kbenoit** [comment]
fcm indexing appears to be failing:

```r
x <- fcm(tokens(c("this contains lots of stopwords",
                  "no if, and, or but about it: lots"),
                remove_punct = TRUE))
x[]
##  Error in x[] : Subscript out of bounds 
```

**Pull request merged**